PyNite/Mesh.py

```diff
@@ -1283,12 +1290,17 @@ def generate(self):
         if num_elements == None:
             num_elements = int(2*pi/mesh_size)
 
+        #Remaining height to be meshed
+        height = h - y                  
+        
+        #Number of times the plate height fits in the remaining unmeshed height, resulting at least one element
+        n_vert = max(int(abs(height)/mesh_size),1)
+        
+        #Element height in the vertical direction
+        h_y = height/n_vert    
+
         # Mesh the cylinder from the bottom toward the top
         while round(y, 10) < round(h, 10):
-            
-            height = h - y                  # Remaining height to be meshed
-            n_vert = int(height/mesh_size)  # Number of times the plate height fits in the remaining unmeshed height
-            h_y = height/n_vert             # Element height in the vertical direction
```

Comment:
I agree with adding the "max" function to ensure that at least one course of plates is being generated.

As long as `h_y` is constant on each iteration, it makes sense to move these variables assignments outside the loop. However, I think it's best to leave the definition of `height`, `n_vert`, and `h_y` in the loop. At some point, I may add control points to the cylinder mesh, allowing `h_y` to vary as the cylinder is generated. This could be useful for hydrostatic loads or changes in shell thickness that cause a discontinuity partway up the cylinder. Right now we're forced to generate two separate cylinders.